src/Workspaces/Core/MSBuild/MSBuild/BuildHostProcessManager.cs

```diff
@@ -197,6 +205,23 @@ private void AppendBuildHostCommandLineArgumentsConfigureProcess(ProcessStartInf
         processStartInfo.RedirectStandardError = true;
     }
 
+    private static void AddArgument(ProcessStartInfo processStartInfo, string argument)
+    {
+        // On .NET Core 2.1 and higher we can just use ArgumentList to do the right thing; downlevel we need to manually
+        // construct the string
+#if NET
+        processStartInfo.ArgumentList.Add(argument);
+#else
+        if (processStartInfo.Arguments.Length > 0)
+            processStartInfo.Arguments += ' ';
+
+        if (argument.Contains(' '))
+            processStartInfo.Arguments += '"' + argument + '"';
+        else
+            processStartInfo.Arguments += argument;
```

Comment:
Ah good question and probably "yes" now. If you're on Linux you're just taking the #if NET path, so the primary need here is if you're on Windows. And there you can't have quotes in file paths. When I wrote this the only thing being passed were file paths, but now since we're passing MSBuild options there's potential escaping too since those could have quotes. Will dig into this.

Reply:
Deferring to follow-up PR.